static/css/custom.css

```diff
@@ -1,54 +1,82 @@
 :root {
+  --step--1: clamp(0.8rem, calc(0.7rem + 0.44vw), 0.95rem);
+  --step-0: clamp(1rem, calc(0.88rem + 0.56vw), 1.19rem);
+  --step-1: clamp(1.25rem, calc(1.09rem + 0.69vw), 1.48rem);
+  --step-2: clamp(1.56rem, calc(1.37rem + 0.87vw), 1.86rem);
+  --step-3: clamp(1.95rem, calc(1.71rem + 1.09vw), 2.32rem);
+  --step-4: clamp(2.44rem, calc(2.14rem + 1.36vw), 2.9rem);
+  --step-5: clamp(3.05rem, calc(2.67rem + 1.69vw), 3.62rem);
+  --space-3xs: clamp(0.25rem, calc(0.21rem + 0.19vw), 0.31rem);
+  --space-2xs: clamp(0.5rem, calc(0.42rem + 0.37vw), 0.63rem);
+  --space-xs: clamp(0.75rem, calc(0.67rem + 0.37vw), 0.88rem);
+  --space-s: clamp(1rem, calc(0.88rem + 0.56vw), 1.19rem);
+  --space-m: clamp(1.5rem, calc(1.29rem + 0.93vw), 1.81rem);
+  --space-l: clamp(2rem, calc(1.75rem + 1.11vw), 2.38rem);
+  --space-xl: clamp(3rem, calc(2.63rem + 1.67vw), 3.56rem);
+  --space-2xl: clamp(4rem, calc(3.5rem + 2.22vw), 4.75rem);
+  --space-3xl: clamp(6rem, calc(5.25rem + 3.33vw), 7.13rem);
   --ifm-color-primary: #0080ff;
+  --ifm-color-primary-dark: #0073e6;
+  --ifm-color-primary-darker: #006dd9;
+  --ifm-color-primary-darkest: #005ab3;
+  --ifm-color-primary-light: #1a8dff;
+  --ifm-color-primary-lighter: #2693ff;
+  --ifm-color-primary-lightest: #4da6ff;
   --ifm-color-secondary: #282c34;
+  --ifm-color-secondary-dark: #24282f;
+  --ifm-color-secondary-darker: #22252c;
+  --ifm-color-secondary-darkest: #1c1f24;
+  --ifm-color-secondary-light: #2c3039;
+  --ifm-color-secondary-lighter: #2e333c;
+  --ifm-color-secondary-lightest: #343944;
+  --ifm-font-size-base: var(--step-0);
+  --ifm-global-spacing: var(--space-s);
+  --ifm-footer-link-color: var(--ifm-color-emphasis-300);
 }
 
-/* custom css */
-h2 {
-  font-size: 40px;
-  margin-top: 60px;
-  margin-bottom: 16px;
+/* Typography styles */
```

Comment:
The refactoring essentially tries to reduce magic numbers as much as possible, and standardise values using the provided variables. Also introducing comments as delimiters for styles specific to certain pages or the overall site such that it is easier to maintain.